Makefile

```diff
@@ -39,6 +39,26 @@ rebuild-all: deps fetch-serverless-custom-file
 serverless-deploy.%: deps fetch-serverless-custom-file
 	$(SLS_BINARY) deploy --stage $*
 
+define GEN_TARGETS
+docker-build-$(platform):
+	@cd builder && docker-compose build $(platform)
+
+build-r-$(platform):
+	@cd builder && R_VERSION=$(R_VERSION) docker-compose up $(platform)
+
+test-r-$(platform):
+	@cd test && R_VERSION=$(R_VERSION) docker-compose up $(platform)
+
+bash-$(platform):
+	docker run -it --rm --entrypoint /bin/bash -v $(CURDIR):/r-builds r-builds:$(platform)
+
+.PHONY: docker-build-$(platform) build-r-$(platform) test-r-$(platform) bash-$(platform)
```

Comment:
This would be better using static pattern rules: https://www.gnu.org/software/make/manual/html_node/Static-Usage.html#Static-Usage

This would allow you to get rid of that `for` loop.

i.e.:
```make
docker-build-$(PLATFORMS):
	@cd builder && docker-compose build $(platform)
```